launcher/ui/pages/modplatform/ResourcePage.cpp

```diff
@@ -324,9 +325,9 @@ void ResourcePage::onSelectionChanged(QModelIndex curr, [[maybe_unused]] QModelI
     updateUi();
 }
 
-void ResourcePage::onVersionSelectionChanged(int index)
+void ResourcePage::onVersionSelectionChanged(int)
 {
-    m_selected_version_index = index;
+    m_selected_version_index = m_ui->versionSelectionBox->currentData().toInt();
```

Comment:
Please test it out yourself, but in my case, it worked as intended.

Reply:
I definitely remember another PR changing something not to use this because of its behaviour...
why not
```c++
m_ui->versionSelectBox->itemData(index)
```

Reply:
I didn't notice any issues with this approach in my testing. Have you actually tested it and found issues?
Or do you have a link to that PR?

Reply:
I can't remember the PR name, but leaving the index unused is kind of pointless

Reply:
Used your suggestion, but from my knowledge and testing the `currentData` does the same thing.
PS. I extended this to all the other QCombobox slots